compiler/rustc_ast/src/mut_visit.rs

```diff
@@ -34,6 +34,10 @@ impl<A: Array> ExpectOne<A> for SmallVec<A> {
     }
 }
 
+pub trait NoopVisitItemKind {
```

Comment:
using `noop` here is confusing to me. This seems to match `SuperVisitable` for types, so I'd really like this to be renamed to sth like `trait SuperVisitableItem`

Reply:
I used the naming convention that is used everywhere else in mut_visit.rs, there are dozens of `noop_visit`s there.
(Same applies to visit.rs.)

Reply:
yeah, i dislike that 😢 i guess it's fine to keep it, but if you have the time, I would be really happy to see it changed in a separate PR then

Reply:
I think it's better to keep the consistent naming, and at some point change the naming scheme in one go (if change at all).

(I don't have any hard feelings towards the current naming personally, it's been around for like 10 years, and at least everyone working with AST knows it.)

Reply:
The style guide recommends against "able" in traits though - https://github.com/rust-lang/rfcs/blob/master/text/0344-conventions-galore.md#trait-naming (at least that's what standard library follows, so I've been always following it in the compiler as well).

Reply:
(Also, this is about `ItemKind` specifically, not `Item`)

Reply:
I guess that's alright 👍 we do use `-able` for quite a few existing traits, esp the existing `ty` visitors, but 🤷